File: Laba1-Python/Task3.py
import math
import Extensions


def Run():
    for i in range(3):
        GetFunctionValue()

def GetFunctionValue():
    x = Extensions.SafeIntInput("Введіть значення аргументу: ")

    if x < -4 or x > 10:
        print("Значення функції немає")
    elif -4 <= x <= -2:
        print(f"Значення функції: {x + 3}")
    elif -2 <= x <= 4:
        print(f"Значення функції: {-x / 2}")
    elif 4 <= x <= 6:
        print("Значення функції: -2")
    elif 6 <= x <= 10:
        result = -2 + math.sqrt(4 - (x - 8) ** 2)
        print(f"Значення функції: {round(result, 3)}")

File: Laba1/Task3.cs
using Extensions;

namespace PZPK_Labas.Laba1
{
    public static class Task3
    {
        public static void Run()
        {
            Console.WriteLine("Завдання 3. Написати програму, яка по веденому значенню аргументу обчислює значення функції, що задається у вигляді графіку.");

            for (int i = 0; i < 3; i++)
                GetFunctionValue();
        }

        private static void GetFunctionValue()
        {
            int x;
            ConsoleInput.SafeIntInput("Введіть значення аргументу", out x);

            if (x < -4 || x > 10)
                Console.WriteLine("Значення функції немає");
            else if (x >= -4 && x <= -2)
                Console.WriteLine($"Значення функції: {x + 3}");
            else if (x >= -2 && x <= 4)
                Console.WriteLine($"Значення функції: {-(x / 2)}");
            else if (x >= 4 && x <= 6)
                Console.WriteLine($"Значення функції: {-2}");
            else if (x >= 6 && x <= 10)
                Console.WriteLine($"Значення функції: {Math.Round(-2 + Math.Sqrt(4 - Math.Pow(x - 8, 2)), 3)}");
        }
    }
}
